apps/ejabberd/src/xmpp_router.erl

```diff
@@ -0,0 +1,15 @@
+-module(xmpp_router).
+-export([route/4]).
+-include_lib("ejabberd/include/ejabberd.hrl").
+
+
+-callback do_route(From :: ejabberd:jid(), To :: ejabberd:jid(), Packet :: jlib:xmlel()) -> ok.
+route(Module,From,To,Packet) ->
+    case (catch Module:do_route(From,To,Packet)) of
+        {'EXIT', Reason} ->
+            ?ERROR_MSG("~p when routing ~ts from ~ts to ~ts in module: ~p",
+                       [Reason, exml:to_binary(Packet),
```

Comment:
I suggest reorganising the log message so that the packet is the last element. It will be more readable. Also I think simple test for the `'EXIT'` case would be good here.